Automattic/jetpack
# Publicize: display the editor modal regardless of module status

Fixes #12367

## Proposed changes:

- Always register the Publicize editor extension.
- Add a new placeholder when the panel is originally loading.
- Add a placeholder with a button to enable the module when it is disabled.
- Track views of that placeholder and clicks to enable the module.

<img width="867" alt="Screenshot 2023-09-18 at 16 41 19" src="https://github.com/Automattic/jetpack/assets/426388/07ce91ac-55c3-44c9-8f5e-24d14a8d13be">
<img width="408" alt="Screenshot 2023-09-18 at 16 40 28" src="https://github.com/Automattic/jetpack/assets/426388/1b399c42-2b11-4137-8616-a55f05cc2473">

### Other information:


## Jetpack product discussion

* #32401 

## Does this pull request change what data or activity we track or use?

* Yes, it starts tracking 2 things: views of the placeholder, and clicks to activate the module.

## Testing instructions:

* Start your test on a site where the Publicize module is not yet enabled.
* Open your browser's JavaScript console, and enable viewing tracks items: `localStorage.setItem( 'debug', 'dops:analytics*' );`
* Go to Posts > Add New
* Click on the Jetpack plugin sidebar.
* Click on the "share this post" panel title.
    * You should see the Tracks event in your JavaScript console.
* Click on the button.
    * You should see the module getting enabled, and the event in your JS console.

You can also test on WordPress.com Simple; the module is always enabled there so the panel should never display the placeholder.
